packages/tools/replay-tool/src/helpers.ts

```diff
@@ -119,16 +125,36 @@ export async function loadContainer(
         summaryOptions: { disableSummaries: true, maxOpsSinceLastSummary: 100000 },
         gcOptions: { writeDataAtRoot: true },
     };
-    const codeLoader = new ReplayCodeLoader(new ReplayRuntimeFactory(runtimeOptions, dataStoreRegistries));
+    const codeLoader = new ReplayCodeLoader(
+        new ReplayRuntimeFactory(runtimeOptions, dataStoreRegistries, requestHandlers));
 
     // Load the Fluid document
     const loader = new Loader({
         urlResolver,
         documentServiceFactory,
         codeLoader,
-        options: {},
+        options: loaderOptions ?? {},
         logger,
     });
 
     return loader.resolve({ url: resolved.url });
 }
+
+export async function containerRuntimeResolver(request: IRequest, runtime: IContainerRuntime){
+    if (request.url === "/containerRuntime") {
+        return { mimeType: "fluid/object", status: 200, value: runtime };
+    } else {
+        return create404Response(request);
+    }
+};
+
+export async function uploadContainerSummary(container: IContainer) {
+    const response = await container.request({ url: "/containerRuntime" })
+    const runtime = response.value as ContainerRuntime;
+    const summaryResult = await runtime.getSummary();
```

Comment:
You can use `summarize` directly right. Why do we need `getSummary`?

Reply:
To run `summarize` directly you need to be aware of garbage collector state (runGC: this.garbageCollector.shouldRunGC,). Does it make sense to move GC params out of summarize options, and let runtime deduce them ?

Reply:
how does ondemand work? my guess is it is abstracted in whatever pattern that is. i'm pretty against increasing our container runtime surface area for this.

Reply:
You can set runGC to true. We want to run GC in the snapshots tests.

Reply:
Alternatively, setting `allowGC` to true in container runtime options will also enable GC.

Reply:
Running with `runGC=true` caused some snapshot tests to fail. Let me figure out which tests were those, then let's chat about more specific issues.

Reply:
Ah, my bad. `runGC=true` will override all configs and run GC. We don't want that. We want GC to run only if the .metadata blob in the snapshot says so. We can modify the default value of `runGC` from `true` to `this.garbageCollector.shouldRunGC` in `summarize` in container runtime.

Reply:
Made the change - also made logger as optional.